lib/ecto/adapters/sqlite3/connection.ex

```diff
@@ -1359,8 +1397,13 @@ defmodule Ecto.Adapters.SQLite3.Connection do
   def expr({fun, _, args}, sources, query) when is_atom(fun) and is_list(args) do
     {modifier, args} =
       case args do
-        [rest, :distinct] -> {"DISTINCT ", [rest]}
-        _ -> {[], args}
+        [_rest, :distinct] ->
+          raise Ecto.QueryError,
+            query: query,
+            message: "Distinct not supported in expressions"
```

Comment:
```sql
sqlite> create table example(a integer);
sqlite> insert into example(a) values(1), (1), (2);
sqlite> select count(distinct a) from example;
2
```

Reply:
It sort of support's them not in the same way that postgres and ecto support them. I started work adding them into this following the mysql ecto adapter approach but threw it out cause it got hairy. Probably worth a second look in a further PR though.